### PR TITLE
Add Map Visualisations from geographical data

### PR DESCRIPTION
**Pull request by lionick**

### CODE REVIEW

File: View/Elements/cou.ctp [hunk 1]
```diff
@@ -20,81 +20,133 @@
...
+                        <input type="text" id="<?php print $prefix ?>DateTo" name="<?php print $prefix ?>DateTo" data-provide="datepicker"/>
+                      <?php if ($prefix == 'registered' || $prefix == 'cou') : ?>
+                          <div class="btn-group">
+                              &nbsp;
+                              <button type="button" class="btn btn-default dropdown-toggle filter-button"
+                                            data-toggle="dropdown">
+                                  Filter <span class="caret"></span>
+                              </button>
+                              <ul class="dropdown-menu">
+                                  <li><a href="#" onclick="return false;" class="groupDataByDate" data-value="daily">Daily
```
Comment (ioigoume) [inline View/Elements/cou.ctp:80]: Implement this with foreach. The values should be enumerators

File: View/Elements/cou.ctp [hunk 2]
```diff
@@ -20,81 +20,133 @@
...
+                  <h3 class="box-title"><?php print _txt('pl.rciamstatsviewer.' . $prefix . '.pl'); ?> </h3>
+              </div>
+              <!-- /.box-header -->
+              <div class="box-body dataTableWithFilter">
+                <?php
+                if ($vv_permissions['registered']) : ?>
+                    <div class="dataTableDateFilter bg-box-silver">
+                        From: &nbsp;
+                        <input type="text" id="<?php print $prefix ?>DateFrom" name="<?php print $prefix ?>DateFrom" data-provide="datepicker"/>
+                        &nbsp;&nbsp;&nbsp;To: &nbsp;
```
Comment (ioigoume) [inline View/Elements/cou.ctp:70]: Consider using `&emsp;` instead of `&nbsp` multiple times

File: View/Elements/dashboard.ctp [hunk 3]
```diff
@@ -1,71 +1,82 @@
...
+  <div class="box" data-type="dashboard">
+    <div class="box-header with-border">
+      <h3 class="box-title"><?php print _txt('pl.rciamstatsviewer.dashboard.logins') ?></h3>
+    </div>
+
+    <div class="box-body dataTableWithFilter">
+      <?php if ($vv_permissions['registered']) : ?>
+        <div class="dataTableDateFilter bg-box-silver">
+          From: &nbsp;<input type="text" id="dashboardDateFrom" name="dashboardDateFrom" data-provide="datepicker" />
+          &nbsp;&nbsp;&nbsp;To: &nbsp;<input type="text" id="dashboardDateTo" name="dashboardDateTo" data-provide="datepicker" />
```
Comment (ioigoume) [inline View/Elements/dashboard.ctp:52]: The same as above

File: View/Elements/dashboard.ctp [hunk 4]
```diff
@@ -1,71 +1,82 @@
...
+          &nbsp;
+          <div class="btn-group">
+            &nbsp;<button type="button" class="btn btn-default dropdown-toggle filter-button" data-toggle="dropdown">
+              Filter <span class="caret"></span>
+            </button>
+            <ul class="dropdown-menu">
+              <li><a href="#" onclick="return false;" class="groupDataByDate" data-value="daily">Daily Basis</a></li>
```
Comment (ioigoume) [inline View/Elements/dashboard.ctp:59]: Use foreach and enumerator or CAKEPHP helper

File: View/Elements/modal.ctp [hunk 5]
```diff
@@ -52,10 +52,12 @@
...
                                 </div>
                                 <div class="box-body dataTableWithFilter">
-                                    <?php if($datatableExport): ?>
+                                    <?php if($datatableExport) : ?>
                                     <div class="dataTableDateFilter bg-box-silver">
-                                        <?php print _txt('pl.rciamstatsviewer.ranges.from'); ?>: &nbsp;<input type="text" id="specificDateFrom" name="specificDateFrom" data-provide="datepicker" />
-                                        &nbsp;&nbsp;&nbsp;<?php print _txt('pl.rciamstatsviewer.ranges.to'); ?>: &nbsp;<input type="text" id="specificDateTo" name="specificDateTo" data-provide="datepicker" />
+                                        <?php print _txt('pl.rciamstatsviewer.ranges.from'); ?>: &nbsp;
+                                        <input type="text" id="specificDateFrom" name="specificDateFrom" data-provide="datepicker" />
+                                        &nbsp;&nbsp;&nbsp;
```
Comment (ioigoume) [inline View/Elements/modal.ctp:59]: same as above

File: View/Elements/tab.ctp [hunk 6]
```diff
@@ -1,91 +1,102 @@
...
-            </div>
-            <!-- ./col -->
-            <div class="col-lg-3 col-xs-6">
-                <!-- small box -->
-            </div>
-            <!-- ./col -->
+  <div class="totalData" id="<?php print Inflector::pluralize($prefix); ?>TotalInfo">
+    <h1><?php print _txt('pl.rciamstatsviewer.' . $prefix . '.pl'); ?></h1>
+    <div class="row">
+      <div class="col-lg-3 col-xs-6">
```
Comment (ioigoume) [inline View/Elements/tab.ctp:5]: Use a foreach for the small boxes

File: View/Elements/tab.ctp [hunk 7]
```diff
@@ -1,91 +1,102 @@
...
+    <div class="box" data-type="<?php print $prefix ?>">
+      <div class="box-header with-border">
+        <h3 class="box-title"><?php print _txt('pl.rciamstatsviewer.' . $prefix . '.pl'); ?> </h3>
+      </div>
+      <!-- /.box-header -->
+      <div class="box-body dataTableWithFilter">
+        <?php if($vv_permissions['registered']) : ?>
+          <div class="dataTableDateFilter bg-box-silver">
+            From: &nbsp;<input type="text" id="<?php print $prefix ?>DateFrom" name="<?php print $prefix ?>DateFrom" data-provide="datepicker" />
+            &nbsp;&nbsp;&nbsp;To: &nbsp;<input type="text" id="<?php print $prefix ?>DateTo" name="<?php print $prefix ?>DateTo" data-provide="datepicker" />
```
Comment (ioigoume) [inline View/Elements/tab.ctp:65]: same as above

File: View/Elements/tab.ctp [hunk 8]
```diff
@@ -1,91 +1,102 @@
...
+              <div class="btn-group">
+                &nbsp;<button type="button" class="btn btn-default dropdown-toggle filter-button" data-toggle="dropdown">
+                  Filter <span class="caret"></span>
+                </button>
+                <ul class="dropdown-menu">
+                  <li><a href="#" onclick="return false;" class="groupDataByDate" data-value="daily">Daily Basis</a></li>
+                  <li><a href="#" onclick="return false;" class="groupDataByDate" data-value="weekly">Weekly Basis</a></li>
+                  <li><a href="#" onclick="return false;" class="groupDataByDate" data-value="monthly">Monthly Basis</a></li>
+                  <li><a href="#" onclick="return false;" class="groupDataByDate" data-value="yearly">Yearly Basis</a></li>
+                </ul>
```
Comment (ioigoume) [inline View/Elements/tab.ctp:77]: same as above.
I noticed that you use the same Dropdown menu multiple times. Perhaps we should move this into its own element

**Review by ioigoume — CHANGES_REQUESTED** [review]

File: View/Elements/cou.ctp (hunk 1 above)
Comment (lionick) [reply View/Elements/cou.ctp]: Done, I have put it to an element also

File: View/Elements/cou.ctp (hunk 2 above)
Comment (lionick) [reply View/Elements/cou.ctp]: Done

File: View/Elements/dashboard.ctp (hunk 3 above)
Comment (lionick) [reply View/Elements/dashboard.ctp]: Done

File: View/Elements/dashboard.ctp (hunk 4 above)
Comment (lionick) [reply View/Elements/dashboard.ctp]: Done, I have put it to an element also

File: View/Elements/modal.ctp (hunk 5 above)
Comment (lionick) [reply View/Elements/modal.ctp]: Done

File: View/Elements/tab.ctp (hunk 6 above)
Comment (lionick) [reply View/Elements/tab.ctp]: Done

File: View/Elements/tab.ctp (hunk 7 above)
Comment (lionick) [reply View/Elements/tab.ctp]: Done

File: View/Elements/tab.ctp (hunk 8 above)
Comment (lionick) [reply View/Elements/tab.ctp]: Done, I have put it to an element also

**Review by ioigoume — APPROVED** [review]
LGTM

**Pull request merged**